File: tests/performance_tests/sieve.py
def sieve(n):
    primes = [True] * (n+1)
    counter = 0
    for i in range(2,n):
        if primes[i]:
            counter = counter + 1
            for j in range(i*i, n, i):
                primes[j] = False

    return counter

if __name__ == "__main__":
    assert sieve(10000000) == 664579

File: tests/performance_tests-py2many/sieve.jl
function sieve(n)::Int64
    primes = repeat([true], (n + 1))
    counter = 0
    for i in (2:n-1)
        if primes[i+1]
            counter = counter + 1
            for j in (i*i:i:n-1)
                primes[j+1] = false
            end
        end
    end
    return counter
end

function main()
    @assert(sieve(10000000) == 664579)
end

main()
